### ISSUE TITLE
[Bug]: Models extending \mixed instead of _BaseCollection

### ISSUE THREAD

**Issue by TWithers** [laravel-idea/plugin#1036]
### Bug description

I ran an update today to the latest version of the plugin today and regenerated the autocomplete and saw errors with model collections in my Laravel 11 apps.

Example: `User::get()->mapWithKeys(...)` shows mapWithKeys underlined with this message: `Method 'mapWithKeys' not found in \App\Models\User[]|\LaravelIdea\Helper\App\Models\_IH_User_C`

I noticed that in Laravel 11, generated models are extending the `\mixed` type:
```php
    class _IH_User_C extends \mixed {
        /**
         * @param int $size
         * @return User[][]|Collection<_IH_User_C>
         */
        public function chunk($size)
        {
            return [];
        }
    }

<?php //c323fa58bef0b31e793109c0c3ad61fc
/** @noinspection all */

namespace  {

    use LaravelIdea\Helper\_BaseCollection;
    
    class mixed extends _BaseCollection {}
}
```

In Laravel 10, I am not seeing the same errors and the generated models extend `_BaseCollection`
```php
    class _IH_User_C extends _BaseCollection {
        /**
         * @param int $size
         * @return User[][]|Collection<_IH_User_C>
         */
        public function chunk($size)
        {
            return [];
        }
    }
```

I set up a new Laravel 11 project to check if I did something strange with my settings and the result was the same. If I clear the generated files, everything works right. As soon as I regenerate the files, it goes back to underlining the code.

Please let me know if it is a bug or if I have some setting completely messed up somewhere.

### Plugin version

8.2.2.241

### Operating system

MacOS

**Comment by adelf** [comment]
Hello.

For some reason, it found that the `newCollection` method of this class returns the `\mixed` type. Do you have a custom `newCollection` method?

**Comment by TWithers** [comment]
This was a fresh Laravel 11 install. Nothing else added.

**Comment by toddstoker** [comment]
I'm experiencing the same behavior.

- MacOS
- PHPStorm 2024.1.4 
- Laravel Idea 8.2.2.241 
- In a Laravel 11 project

**Comment by adelf** [comment]
Fixed in the Laravel Idea 8.2.3.
